src/map.rs

```diff
@@ -166,7 +167,31 @@ pub struct TryInsertError<'a, V> {
     pub old: &'a V,
     /// A reference to the new value, the one passed as an argument to
     /// [`HashMap::try_insert`].
-    pub not_inserted: &'a V,
+    pub not_inserted: V,
+}
+
+impl<'a, V> Display for TryInsertError<'a, V>
+where
+    V: Display,
+{
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "The value \"{}\" was inserted into the map, but the value \"{}\" was already associated with that key", 
```

Comment:
This seems a little misleading, since `self.not_inserted` was, well, _not_ inserted.

Reply:
> This seems a little misleading, since `self.not_inserted` was, well, _not_ inserted.

Indeed! I'm not very good at writing friendly error messages, could you help me out in this one?

Also, I think we should implement `Display` regardless of whether or not `V` implements `Display`. My idea is to provide a generic error message that doesn't take into consideration the `old` and `not_iserted` fields.If the user want's to take those fields into account, he can always access them and create a custom error message.

Reply:
How about:
> Insert of "{}" failed as key was already present with value "{}"

I think I would probably implement `Display` if `V: Debug`. That seems like a reasonable compromise to me.